### ISSUE TITLE
Python 3.11 support

### ISSUE THREAD

**Issue by ndevenish** [dls-controls/scanspec#57]
Unfortunately it looks like apischema (no updates since march, looks relatively dead?) is [not compatible with python 3.11](https://github.com/wyfo/apischema/issues/485), because it's using private APIs:
```
>>> import scanspec
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/env/lib/python3.11/site-packages/scanspec/__init__.py", line 1, in <module>
    from . import regions, specs
  File "/env/lib/python3.11/site-packages/scanspec/regions.py", line 7, in <module>
    from apischema import schema
  File "/env/lib/python3.11/site-packages/apischema/__init__.py", line 30, in <module>
    from . import (  # noqa: F401
  File "/env/lib/python3.11/site-packages/apischema/cache.py", line 6, in <module>
    from apischema.utils import type_dict_wrapper
  File "/env/lib/python3.11/site-packages/apischema/utils.py", line 37, in <module>
    from apischema.typing import (
  File "/env/lib/python3.11/site-packages/apischema/typing.py", line 98, in <module>
    from typing import _collect_type_vars  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name '_collect_type_vars' from 'typing' (/env/lib/python3.11/typing.py)
>>>
```

**Comment by coretl** [comment]
Depending on apischema now looks like a mistake, @callumforrester is now seeing how painful it would be to move to pydantic instead in #56

**Comment by DominicOram** [comment]
We would like to get this fixed soon so that we can move https://github.com/dials/nexgen to Python 3.11. How far along are we with moving to `pydantic`, anything we can do to help?

**Comment by coretl** [comment]
@callumforrester what do you reckon?

**Comment by callumforrester** [comment]
@coretl @DominicOram WIP here: https://github.com/dls-controls/scanspec/pull/56 
Hoping to get back to this soon, current challenge is in making a tagged union of subclasses, see https://github.com/pydantic/pydantic/discussions/4706. If you guys have any ideas RE that discussion, that would be helpful!

**Comment by wyfo** [comment]
> Unfortunately it looks like apischema (no updates since march, looks relatively dead?)

Sorry, I took a (long) break. Python3.11 support coming soon (tonight normally).

**Comment by coretl** [comment]
@callumforrester released 0.5.5 which supports python 3.11
